hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```diff
@@ -120,6 +121,7 @@ private void validateIdentifier(String id, Set<String> identifiers, String confi
 
   private StructType getExpectedTransformedSchema(TransformerInfo transformerInfo, JavaSparkContext jsc, SparkSession sparkSession,
                                                   Option<StructType> incomingStructOpt, Option<Dataset<Row>> rowDatasetOpt, TypedProperties properties) {
+    Option<Schema> sourceSchemaOpt = sourceSchemaSupplier.get();
```

Comment:
Could you add a test of a scenario where the schema is evolved from the schema provider, and the `getExpectedTransformedSchema` returns the updated `StructType` instance?

Reply:
added a test to validate that the supplier is called per invocation of the method